pkg/api/validation/validation.go

```diff
@@ -2589,6 +2610,10 @@ func ValidatePodUpdate(newPod, oldPod *api.Pod) field.ErrorList {
 	mungedPod.Spec.Tolerations = oldPod.Spec.Tolerations
 	allErrs = append(allErrs, validateOnlyAddedTolerations(newPod.Spec.Tolerations, oldPod.Spec.Tolerations, specPath.Child("tolerations"))...)
 
+	// munge spec.Priority*
+	mungedPod.Spec.PriorityClassName = oldPod.Spec.PriorityClassName
```

Comment:
I guess you are referring to the code at line 2576-2580. Those lines verify that if the class name has changed and the new name is not empty, the new name is a valid name.